presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -109,144 +119,138 @@ public VerificationContext getVerificationContext()
     public VerificationResult run()
     {
         boolean resultMismatched = false;
-        QueryBundle control = null;
-        QueryBundle test = null;
-        MatchResult matchResult = null;
-        Optional<DeterminismAnalysis> determinismAnalysis = Optional.empty();
-
-        QueryStats controlQueryStats = null;
-        QueryStats testQueryStats = null;
 
+        AtomicReference<QueryBundle> control = new AtomicReference<>();
```

Comment:
What's the purpose of using AtomicReferences?

Reply:
In 141 - 150, those variables are used in lambda, and IntelliJ complains about "Variable used in lambda expression should be final or effectively final" if they're `nullable` or `Optional`.

Instead, I need a reference to those variables, with setter method for updating the value. `AtomicReference` seems to be easiest to use although I don't need the atomicity of the update. Otherwise, I'll need create a private static inner class `Reference`.

Reply:
That seems sketchy to me, but I can't tell if it's actually a problem here or just seems bad (https://www.baeldung.com/java-lambda-effectively-final-local-variables is what I read to understand what the problems might be).  I feel it would be better to change it because my general philosophy is it's best if a reader doesn't have to think too hard to understand why code is correct.  However, you could alternatively get a review from someone with deeper Java knowledge to say whether there are any concerns in this case, and then also add a comment about why these use AtomicReference and why it's not a problem.

Reply:
This is an annoying limitations, but I usually create copy variables like so:

```
QueryBundle control = null;

        try {
            // Rewrite queries
            control = queryRewriter.rewriteQuery(sourceQuery.getControlQuery(), CONTROL);
            test.set(queryRewriter.rewriteQuery(sourceQuery.getTestQuery(), TEST));

            // Run queries
            QueryBundle controlCopy = control;
            runAndConsume(
                    () -> setupAndRun(prestoAction, controlCopy, false),
                    controlStats::set,
                    e -> controlState.set(getFailingQueryState(e)));
```

Reply:
@mbasmanova 
Copy would work only for variables to be read, but I also need values to be set in the the lambda.
```
controlStats::set,
e -> controlState.set(getFailingQueryState(e))
```
controlStats and controlState are used as a 1-field mutable context object. It is similar to creating a private mutable Context object wrapping all those field, define setters on each of them, and supply `context::setControlStats` and `e->context.setControlState(getFailingQueryState(e))`.

Reply:
@caithagoras Thanks for clarifying. I like the solution using Context object better.

Reply:
Done. Since @rschlussel advices against mutable object, I'm making the mutable context object a private static inner class to minimize exposure.